### ISSUE TITLE
Zoom in/out when using zones not working with boost

### ISSUE THREAD

**Issue by Abrissirba** [highcharts/highcharts#9124]
#### Expected behaviour
When zooming in and out I expect to only have one line per serie

#### Actual behaviour
If I zoom in and out multiple times I sometimes get two lines for a serie. This happens when I use zones together with the boost module.

#### Live demo with steps to reproduce
http://jsfiddle.net/81vhr764/5/

The fiddle shows a use case where I want multiple axes with one serie per axis stacked in the same graph. I use a zone for each graph to hide any points that lays below the axis min value. If I zoom in and out multiple times, sometimes the bottom serie get multiple lines visible in the graph. 

#### Product version
Highcharts 6.1.1

#### Affected browser(s)
Tested in chrome

**Comment by pawelfus** [comment]
Hi @Abrissirba 

Thank you for reporting the issue! You are right, it looks broken.

**Workaround:**
Use only boosted rendering engine by setting [`boost.seriesThreshold=1`](https://api.highcharts.com/highstock/boost.seriesThreshold).

Demo: http://jsfiddle.net/BlackLabel/ks9zqc07/2/

**Simplified demo:**
http://jsfiddle.net/BlackLabel/ks9zqc07/

*Internal note:*
It's probably caused by generating separate shapes for each zone and boost is not clearing them all.

**Comment by yoderexploder** [comment]
I'm interested in this as well, and I see you committed changes referencing this issue.  Is there anything that needs to be done to get zones working with Boost?  Thanks so much.

**Comment by pawelfus** [comment]
Hi @yoderexploder 

My demo above seems to be fine ( http://jsfiddle.net/BlackLabel/ks9zqc07/ ) - could you recreate issue in jsFiddle? In case something is still broken, please consider using my workaround posted above, it should help.

**Comment by yoderexploder** [comment]
Thanks @pawelfus, for your response.  

I think I've just figured out why my zones were not working the Boost module.  It looks as if you must define a 'value' for every zone.  This is in contrast to not using Boost where the value for the final zone does not need to be defined.  I've worked around this by just setting the value of the last zone to Infinity.  Example here: [https://jsfiddle.net/09cxhn7f/](https://jsfiddle.net/09cxhn7f/)

I am still having a problem after resetting the zoom.  The zoomed image remains after zooming back out.  The same fiddle demonstrates that problem.  I think you've got an issue logged for this already, i'll see if I can find it and participate there.
